### PR TITLE
fix typed parameters in access path

### LINKED ISSUE #1000: regression on fully-qualified names
When one spoon https://github.com/spoon-bot/guava with the latest snapshot, `AbstractMapBasedMultimap.java` is not correctly rewritten.

     /builds/workspace/Guava/guava/target/generated-sources/spoon/com/google/common/collect/AbstractMapBasedMultimap.java:[656,16] error: AbstractMapBasedMultimap.WrappedList.WrappedListIterator is not abstract and does not override abstract method next() in ListIterator

@pvojtechovsky it's maybe due to your work on access path.

### PR DESCRIPTION
**Pull request by pvojtechovsky**
fixes #1000

### CODE REVIEW

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 1]
```diff
@@ -324,6 +324,9 @@ public TypeFactory() {
 		}
 
 		ref.setSimpleName(type.getSimpleName());
+		for (CtTypeParameter typeParam : type.getFormalCtTypeParameters()) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]: why is this change required?

Comment (pvojtechovsky) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]: I have already found, that this change is wrong. I will rollback it and do it different.

**Comment by pvojtechovsky** [comment]
This implementation is still not 100% correct, but 
- it is fitting more to origin concept of @tdurieux, because CtTypeReference.toString() now returns access path and not qualified name - same like in his origin concept! I restored some test cases to the state before my accessPath implementation.
- all tests pass and guava seems to compile now too.

The remaining problem is how to correctly map actual type arguments. See the comment in `CtTypeReferenceImpl.applyActualTypeArguments`.
I do not want to implement that algorithm now. I vote for merging it as it is and to fix it after somebody writes code which will really need that. ... the origin problem was very rare and the case when this implementation fails should be even more rare :-)

**Comment by monperrus** [comment]
it looks good to me, thanks. @tdurieux OK for you?

> CtTypeReference.toString() now returns access path and not qualified name - same like in his origin concept!

this is fine, in general `toString` on references is not as important as `toString` on code elements, it's more for debug purposes, so the specification is more malleable.

**Comment by tdurieux** [comment]
@monperrus it is good for me

**Pull request merged**